django_multitenant/mixins.py

```diff
@@ -69,11 +70,21 @@ def _do_update(self, base_qs, using, pk_val, values, update_fields, forced_updat
                                                        forced_update)
 
     def save(self, *args, **kwargs):
+        current_tenant = get_current_tenant()
         tenant_value = get_current_tenant_value()
+
         if self.tenant_value is None and tenant_value and not isinstance(tenant_value, list):
             setattr(self, self.tenant_field, tenant_value)
 
-        return super(TenantModelMixin, self).save(*args, **kwargs)
+        if self.tenant_value and tenant_value != self.tenant_value:
+            self_tenant = get_object_tenant(self)
+            set_current_tenant(self_tenant)
+
+        obj = super(TenantModelMixin, self).save(*args, **kwargs)
+
+        set_current_tenant(current_tenant)
```

Comment:
Hi @louiseGrandjonc , I upgraded to the latest version (v2.3.0) and ran into an issue due to this change. More specifically, if the [save to DB at line 83](https://github.com/citusdata/django-multitenant/pull/88/files#diff-6affa3189b6033e7615a6bc23fb18f8eR83) fails (e.g. because of uniqueness validation), this `set_current_tenant` would not be executed and the code would continue with a "wrong" tenant ([the tenant of the object being saved](https://github.com/citusdata/django-multitenant/pull/88/files#diff-6affa3189b6033e7615a6bc23fb18f8eR81)). I wonder if we could wrap the `save` operation in a try-finally block so that we always reset the current tenant? for example
```python
try:
    obj = super(TenantModelMixin, self).save(*args, **kwargs)
finally:
    set_current_tenant(current_tenant)
```

Reply:
That's a brilliant idea, I'll do that now.

Reply:
I just tested `v2.3.2` and it worked like a charm. Thank you very much for your amazingly quick response (as well as this awesome library)!